sdk/dotnet/Pulumi/Serialization/ResourcePackages.cs

```diff
@@ -35,7 +35,7 @@ internal static bool TryGetResourceType(string name, string? version, [NotNullWh
                 _resourceTypes ??= DiscoverResourceTypes();
             }
 
-            var minimalVersion = version != null ? SemVersion.Parse(version) : new SemVersion(0);
+            var minimalVersion = !string.IsNullOrEmpty(version) ? SemVersion.Parse(version) : new SemVersion(0);
```

Comment:
Otherwise `SemVersion.Parse(version)` was throwing an exception when it is passed an empty string:

https://github.com/pulumi/pulumi/blob/d926fe6c1369da41787b3d3cac40c73aa98f4a2b/sdk/dotnet/Pulumi/Serialization/Deserializer.cs#L227-L230